.update.R

```diff
@@ -0,0 +1,69 @@
+R_pkgs <- c('base', 'compiler', 'datasets', 'graphics', 'grDevices', 'grid', 'methods', 'parallel', 'splines', 'stats', 'stats4', 'tcltk', 'tools', 'utils') #rownames(utils::installed.packages(grep('lib-R', packrat:::getLibPaths(), value=TRUE)))
+rocker_pkgs <- c(ext_pkgs, R_pkgs)
+
+# Create additional symlinks from the system libraries to the packrat/lib
+# directory for those packages that are dependencies of GRAN or GitHub packages.
+# This helps with packrat::snapshot(), which throws errors if there are packages
+# in packrat/lib that depend on packages only available in packrat/lib-ext or
+# packrat/lib-R (the system libs). So that we don't have to re-run this every
+# time we add a package, just copy all rocker_packages
+ext_details <- utils::installed.packages(packrat:::getDefaultLibPaths())
+pack_lib <- grep('/lib/', packrat:::getLibPaths(), value=TRUE)
+success <- sapply(rocker_pkgs, function(pkg) {
+  source <- file.path(ext_details[pkg,'LibPath'], pkg)
+  target <- file.path(pack_lib, pkg)
+  if(!file.exists(target)) {
+    message('symlinking ', pkg)
+    succeeded <- packrat:::ensurePackageSymlink(source, target) # produces T/F about success
+    if(!succeeded) {
+      stop('failed to symlink ', pkg)
+    }
+  } else {
+    return(NA)
+  }
+})
+
+# Run this after you've installed+libraried any new packages
+packrat::snapshot()
+
+# Once you're done adding packages and snapshotting, remove those symlinks
+success <- sapply(rocker_pkgs, function(pkg) {
+  symlink <- file.path(pack_lib, pkg)
+  is.symlink <- !is.na(Sys.readlink(symlink)) && (Sys.readlink(symlink) != "")
+  if(is.symlink) {
+    message('removing symlink for ', pkg)
+    unlink(symlink)
+  }
+})
+
+# Amend the DESCRIPTION files of GRAN packages to include a Repository line.
+# This is needed to avoid "Error: Unable to retrieve package records" during
+# packrat::restore()
+library(dplyr)
+gran_pkgs <- c('dataRetrieval','grithub')
```

Comment:
this is a manual list that needs editing for now. if everything else works, i'll want to figure out how we can identify these packages in a more automated way. And/or work with @ldecicco-USGS to see if we can get GRAN to insert a `Repository` line in the description files.

Reply:
hmm....sure? Want to make an issue in USGS-R/grantools?

Reply:
Done: https://github.com/USGS-VIZLAB/gage-conditions-docker/pull/1#discussion_r209350609